app/models/taxon.rb

```diff
@@ -1645,6 +1645,14 @@ def descendant_of?( taxon )
     ancestor_ids.include?( taxon.id )
   end
 
+  def sibling_of?( taxon )
```

Comment:
I'd change the name of this method, since I think it would be common to think of "siblings" as only the taxa that share the same parent, not any two taxa with disjoint ancestries. Maybe invert some of the logic and call it `in_same_branch_of?` or something like that? A taxon is in the same branch of another taxon if one is an ancestor of the other, or one is a descendant of the other